pkg/apis/garden/types.go

```diff
@@ -1302,6 +1302,106 @@ type KubeletConfig struct {
+	// +optional
+	// Default: 0 for each resource
+	EvictionMinimumReclaim *KubeletConfigEvictionMinimumReclaim
+	// EvictionPressureTransitionPeriod is the duration for which the kubelet has to wait before transitioning out of an eviction pressure condition.
+	// +optional
+	// Default: 4m0s
+	EvictionPressureTransitionPeriod *metav1.Duration
+	// EvictionMaxPodGracePeriod describes the maximum allowed grace period (in seconds) to use when terminating pods in response to a soft eviction threshold being met.
+	// +optional
+	// Default: 90
+	EvictionMaxPodGracePeriod *int32
+}
+
+// KubeletConfigEviction contains kubelet eviction thresholds supporting either a resource.Quantity or a percentage based value.
+type KubeletConfigEviction struct {
+	// MemoryAvailable is the threshold for the free memory on the host server.
+	// +optional
+	MemoryAvailable *string
+	// ImagefsAvailable is the threshold for the free disk space in the imagefs filesystem (docker images and container writable layers).
+	// +optional
+	ImagefsAvailable *string
+	// ImagefsInodesFree is the threshold for the available inodes in the imagefs filesystem.
+	// +optional
+	ImagefsInodesFree *string
+	// NodefsAvailable is the threshold for the free disk space in the nodefs filesystem (docker volumes, logs, etc).
+	// +optional
+	NodefsAvailable *string
+	// NodefsInodesFree is the threshold for the available inodes in the nodefs filesystem.
+	// +optional
+	NodefsInodesFree *string
+}
+
+// KubeletConfigEviction contains configuration for the kubelet eviction minimum reclaim.
+type KubeletConfigEvictionMinimumReclaim struct {
+	// MemoryAvailable is the threshold for the memory reclaim on the host server.
+	// +optional
+	MemoryAvailable *resource.Quantity
+	// ImagefsAvailable is the threshold for the disk space reclaim in the imagefs filesystem (docker images and container writable layers).
+	// +optional
+	ImagefsAvailable *resource.Quantity
+	// ImagefsInodesFree is the threshold for the inodes reclaim in the imagefs filesystem.
+	// +optional
+	ImagefsInodesFree *resource.Quantity
+	// NodefsAvailable is the threshold for the disk space reclaim in the nodefs filesystem (docker volumes, logs, etc).
+	// +optional
+	NodefsAvailable *resource.Quantity
+	// NodefsInodesFree is the threshold for the inodes reclaim in the nodefs filesystem.
+	// +optional
+	NodefsInodesFree *resource.Quantity
+}
+
+// KubeletConfigEvictionSoftGracePeriod contains grace periods for kubelet eviction thresholds.
+type KubeletConfigEvictionSoftGracePeriod struct {
+	// MemoryAvailable is the grace period for the MemoryAvailable eviction threshold.
+	// +optional
+	MemoryAvailable *metav1.Duration
+	// ImagefsAvailable is the grace period for the ImagefsAvailable eviction threshold.
+	// +optional
+	ImagefsAvailable *metav1.Duration
```

Comment:
```suggestion
	ImageFSAvailable *metav1.Duration
```

(rename all accordingly)